src/users/task/UserProcessingDemonstrator.java

```diff
@@ -0,0 +1,53 @@
+package users.task;
+
+import functionalinterface.example.Randomable;
+
+import java.time.Duration;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+
+public class UserProcessingDemonstrator {
+
+    public static void demonstrateUserProcessing() {
+
+        List<User> users = generateUsers(99);
+
+        Map<Team, List<String>> groupedEmails = users.stream()
+                .filter(u -> Duration.between(LocalDate.now().atStartOfDay(), u.getLoginDate().atStartOfDay()).toDays() >= -7)
```

Comment:
`-7` of what? move to constants

Reply:
`LocalDate.now().atStartOfDay()` create separate variable for this

Reply:
`u.getLoginDate().atStartOfDay()` as previous

Reply:
`Duration.between().toDays()` is too complex

use methods from `LocalDate` instead of this

Reply:
fixed